development-strategy.md

```diff
@@ -143,9 +175,22 @@ Ideas about what the page should have..
 * A registered user can reach his results/scores.
 
 Ideas about what the page should have..
-- a user can see test result at the end of each  test and he can also see his wrong questions with the correct answers. 
-- user can view his result after he complete the test
-- user can see the solution/corrections
+- a user can see the the solution/corrections at the end of each test
+- a user can view his result after he complete the test.
+
+__FRONTEND__
+
+* results/statistics button mentioned above. these buttons are in two sections , one is in the test page of the student user  and other is in the main page of the admin.
+* as a student in the test page user see the amount of correct answers for each test besides the test name such as `test-1 21/25`
+* when a user solve a test , after the test he can see the results.
+
+__BACKEND__
+
+* to show the correct answer amount for each test, it `render` his answers with the correct answers and save this statistics to `db`
+* when the test page is loaded , statistics is shown on beside it for each test. 
+* while solving a test after any question , the answer sends to `db`. there it `rendered` and the result table changes.
```

Comment:
For my opinion, it should be at the end of the exam (after 20x4 part), because whole test has time restriction so the student go back and change his answer, it shouldn't make server busy all of the time.

Reply:
ok